components/GardenCards.tsx

```diff
@@ -7,29 +7,11 @@ import { GardenTabScreenProps } from "../types";
 import { View } from "./Themed";
 
 type GardenCardsProps = {
-  setSelectedGardenId: React.Dispatch<React.SetStateAction<string>>;
   navigation: GardenTabScreenProps<"GardenTabScreen">["navigation"];
-  selectedGardenId: string;
 };
 
-export const GardenCards = ({
-  setSelectedGardenId,
-  navigation,
-  selectedGardenId,
-}: GardenCardsProps) => {
+export const GardenCards = ({ navigation }: GardenCardsProps) => {
   const gardens = useAppSelector(gardenSelectors.selectGardens);
-  const setSelectedGarden = setSelectedGardenId;
-
-  const handlePress = async (id: string) => {
```

Comment:
Have a look through this commit a bit, I've removed quite a bit of code by passing the ids with navigate. let me know if I've removed something you needed for something else

Reply:
That looks spot on to me, I was going to have to bring up how best to handle the id's this week anyway and I think this solves it.